# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.h
# Commit: 1034aeb0f8e2
# Full Hash: 1034aeb0f8e2d86ea04d8e6a8c717f13c9d772eb
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-11-18 15:46:32
# Regressor Bug: 1800496
# File Overlap Count: 2
# Description:
#   Bug 1800496 - Revert mCacheCreator to being defined as part of ScriptLoaderRunnable; r=asuth
#   
#   Previously, in the cancellation refactor, we removed knowledge of the list of scripts being loaded on the main thread. As a result of this, we could no longer clear the cache creator on the main thread after iterating over all requests. We can now do that once again, and it simplifies the code.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D162220
# ==============================================================================

diff -r dfe78c41e4eb -r 1034aeb0f8e2 dom/workers/ScriptLoader.h
--- a/dom/workers/ScriptLoader.h	Fri Nov 18 10:02:28 2022 +0000
+++ b/dom/workers/ScriptLoader.h	Fri Nov 18 10:02:28 2022 +0000
@@ -271,6 +271,7 @@
   RefPtr<ThreadSafeWorkerRef> mWorkerRef;
   nsTArrayView<RefPtr<ThreadSafeRequestHandle>> mLoadingRequests;
   Maybe<nsresult> mCancelMainThread;
+  RefPtr<CacheCreator> mCacheCreator;
 
  public:
   NS_DECL_THREADSAFE_ISUPPORTS
@@ -294,6 +295,8 @@
 
   void CancelMainThreadWithBindingAborted();
 
+  CacheCreator* GetCacheCreator() { return mCacheCreator; };
+
  private:
   ~ScriptLoaderRunnable() = default;
 